src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/UseValidPlatformString.cs

```diff
@@ -0,0 +1,176 @@
+                {
+                    return;
+                }
+
+                var knownPlatforms = context.Options.GetMSBuildItemMetadataValues(MSBuildItemOptionNames.SupportedPlatform, context.Compilation, context.CancellationToken);
+                knownPlatforms = knownPlatforms.AddRange(GetPlatformNamesFromGuardMethods(operatingSystemType));
+
+                context.RegisterOperationAction(context => AnalyzeOperation(context.Operation, context, knownPlatforms), OperationKind.Invocation);
+                context.RegisterSymbolAction(context => AnalyzeSymbol(context.ReportDiagnostic, context.Symbol,
+                    supportedAttriubte, unsupportedAttribute, knownPlatforms, context.CancellationToken), s_symbols);
+                context.RegisterCompilationEndAction(context => AnalyzeSymbol(context.ReportDiagnostic, context.Compilation.Assembly,
+                    supportedAttriubte, unsupportedAttribute, knownPlatforms, context.CancellationToken));
+            });
+
+            static IEnumerable<string> GetPlatformNamesFromGuardMethods(INamedTypeSymbol operatingSystemType)
+            {
+                return operatingSystemType.GetMembers().OfType<IMethodSymbol>().Where(m =>
+                        m.IsStatic &&
+                        m.ReturnType.SpecialType == SpecialType.System_Boolean &&
+                        NameAndParametersValid(m)).Select(m => m.Name[2..]);
+            }
+
+            static bool NameAndParametersValid(IMethodSymbol method) =>
+                method.Name.StartsWith(IsPrefix, StringComparison.Ordinal) &&
+                method.Parameters.Length == 0;
+        }
+
+        private static void AnalyzeOperation(IOperation operation, OperationAnalysisContext context, ImmutableArray<string> knownPlatforms)
+        {
+            if (operation is IInvocationOperation invocation &&
+                methodNames.Contains(invocation.TargetMethod.Name) &&
+                invocation.Arguments.Length > 0 &&
+                invocation.Arguments[0].Value is { } argument &&
+                argument.ConstantValue.HasValue &&
+                argument.ConstantValue.Value is string platformName &&
+                IsNotKnownPlatform(knownPlatforms, platformName))
+            {
+                context.ReportDiagnostic(argument.Syntax.CreateDiagnostic(UnknownPlatform, platformName));
+            }
+        }
+
+        private static void AnalyzeSymbol(Action<Diagnostic> reportDiagnostic, ISymbol symbol, INamedTypeSymbol supportedAttrbute,
+            INamedTypeSymbol unsupportedAttribute, ImmutableArray<string> knownPlatforms, CancellationToken token)
+        {
+            foreach (var attribute in symbol.GetAttributes())
+            {
+                if (supportedAttrbute.Equals(attribute.AttributeClass.OriginalDefinition) ||
```

Comment:
I think this should use SymbolEqualityComprare.
Currently RS1024 is disabled due to a codefix bug, usually it reports these.

Reply:
> I think this should use SymbolEqualityComprare.

Is that a type? I cannot find that in the repo

Reply:
never mind, there was a typo, found it